doc/chassis-http-server/chassis_http_server.md

```diff
@@ -0,0 +1,112 @@
+
+
+### Scope  
+
+ This document provides design details for the http-server for internal midplane in Sonic Chassis system.
+
+### Definitions/Abbreviations
+
+### Overview 
+
+ On a Sonic Chassis system, currently there is no support for hosting image and configuration files at a Centralized internal location, i.e., Supervisor, for sharing the same with the Line Card modules. Without this support, each Line Card modules must reach to external servers on the mgmt interface network to download these files.
+
+ The http-server implementation described in this document provides a way to run http-server on the supervisor to host image and configuration files for Line Card module and share those on the internal midplane network. This http-server implementation is not to service the requests on the external network.
+
+
+### Requirements 
+
+ The http-server shall run on the internal midplane network and use the internal Supervisor midplane IP, provided by the platform, to bind.
+ The http-server shall use port 8000 to bind on the internal network.
+
+ The use of this http-server shall be limited to internal midplane and not designed to service requests on the mgmt/external interface.
+
+### Architecture Design 
+
+ The new http-server shall use the current Sonic architecture and shall not require any architecture design change. The server shall run only on the supervisor and inside the existing database-chassis docker, which runs only on supervisor.
+
+ The server shall use the existing redis_chassis.server IP address provided by platform configuration, which is well known IP address used by Line Cards to reach the supervisor module.
+
+ A high-level architecture  design of Chassis http server is as follows:
+
+![Chassis http-server view](chassis-http-server.jpg)
+
+### High-Level Design 
+
+ This service runs in database-chassis docker based on the configuration provided in "device/\<platform\>/chassisdb.conf"
+
+ The existing python http-server module is used to provide the service. As this module is already packaged with python, no new module packaging is required.
+ 
+ Configuration parameters provided by chassisdb.conf are as follows:
+ ```
+  start_http_server: if "1", http-server will be started on this node.
+  chassis_db_address   : IP address for http-server. This is the same IP used for the chassis db server
+  http_server_port : Port to bind to, default: 8000
+  http_server_dir  : HTTP server home directory path, default: /var/www/
+ ```
+
+ **Example of config in chassisdb.conf:**
+ ```
+  start_http_server=1
+  chassis_db_address=127.0.0.10
+  http_server_dir=/var/www/tftp
+ ```
+
+ The service will be added as a new process in docker database-chassis. dockers/docker-database/supervisord.conf.j2 will be modified to support the new service. This service should run only on database-chassis server, which also hosts redis_chassis.server. Other database dockers should not run the http-server.
```

Comment:
not sure why re-use database docker not starting its own docker for this http_server?

Reply:
Hi @lguohan , this feature was discussed in Chassis community group last year and adding http-server docker was one of the options. But as this is a light weight python http-server, community was more in favor of adding a host side http-server feature. A PR was raised for that but that was an issue enabling the http-server on host side as a feature as it required the service to run a docker.

Later in the community discussion it was decided as its needed in 202205 and only for chassis, to add it as a light weight service in database-chassis which only runs for chassis.
This service is only required for Chassis, binds only to midplane/internal network and uses redis_chassis.server IP to bind to host the server.

@abdosi , please add if I missed anything.

Let me know if we need to have a quick meeting to close on this. 
Thanks